aten/src/ATen/core/Tensor.h

```diff
@@ -160,8 +160,8 @@ class CAFFE2_API Tensor {
   /// Returns a `Tensor`'s layout. Defined in Type.h
   Layout layout() const noexcept;
 
-  /// Returns a `Tensor`'s dtype (`ScalarType`). Defined in Type.h
-  ScalarType dtype() const noexcept;
+  /// Returns a `Tensor`'s dtype (`TypeMeta`). Defined in TensorMethods.h
+  const caffe2::TypeMeta& dtype() const noexcept;
```

Comment:
There's no need to return a reference now; it's just a pointer; return it by value.